dissect/target/plugins/os/windows/defender.py

```diff
@@ -108,6 +418,92 @@ def evtx(self) -> Generator[Record, None, None]:
+                    }
+                )
+                yield DefenderFileQuarantineRecordDescriptor(**fields, _target=self.target)
+            else:
+                self.target.log.warning(f"Unknown Defender Detection Type {self.detection_type}")
+
+    @plugin.arg(
+        "--output",
+        "-o",
+        dest="output_dir",
+        type=Path,
+        required=True,
+        help="Path to recover quarantined file to.",
+    )
+    @plugin.export(output="none")
+    def recover(self, output_dir: Path) -> None:
+        if not output_dir.exists():
+            raise ValueError("Output directory does not exist.")
+        quarantine_directory = self.target.fs.path(DEFENDER_QUARANTINE_FOLDER_PATH)
+        resourcedata_directory = quarantine_directory.joinpath(QUARANTINE_RESOURCEDATA_FOLDER_NAME)
+        if resourcedata_directory.exists() and resourcedata_directory.is_dir():
+            recovered_files = []
+            for entry in self.get_quarantined_entry_resources():
+                if entry.detection_type != b"file":
+                    continue
+                subdirectory = resourcedata_directory.joinpath(entry.resource_id[0:2])
+                if not subdirectory.exists():
+                    self.target.log.warning(f"Could not find a ResourceData subdirectory for {entry.resource_id}")
+                    continue
+
+                resourcedata_location = None
+
+                # Sometimes, the resourcedata file containing the quarantined file does not have the exact same name
+                # as the entry's resource_id. Instead, it only matches a part of the resource_id. What we do is loop
+                # over all files in the resourcedata subdirectory, and check whether we can find a filename that
+                # fully fits into the resource_id. If so, we assume that that is the matching file and break.
+                for possible_file in subdirectory.glob("*"):
+                    _, _, filename = str(possible_file).rpartition("/")
+                    if filename in entry.resource_id:
+                        resourcedata_location = resourcedata_directory.joinpath(entry.resource_id[0:2]).joinpath(
+                            filename
+                        )
+                        break
+                if resourcedata_location is None:
+                    self.target.log.warning(f"Could not find a ResourceData file for {entry.resource_id}.")
+                    continue
+                if resourcedata_location in recovered_files:
+                    # We already recovered this file
+                    continue
+                fh = resourcedata_location.open()
+                # TODO: What filename do we want for recovery? Detection path seems OK but what if different files
+                # have the same filename but are stored in different directories? Resource id seems most 'truthful'
+                # but might be confusing for analysts.
```

Comment:
I think resource ID is the most "safe" option here, and also allows for easy cross reference lookups. Analysts should dump the quarantine records to cross reference for more information.